src/preamble.js

```diff
@@ -2401,7 +2403,13 @@ function integrateWasmJS(Module) {
   Module['asmPreload'] = Module['asm'];
 
   // Memory growth integration code
+
+  var defaultReallocBuffer = Module['reallocBuffer'];
+
   Module['reallocBuffer'] = function(size) {
+    if (finalMethod === 'asmjs') {
```

Comment:
Oh right. Makes sense. Although I think `binaryenMethod` might still be better? No matter either way, lgtm to merge now, thanks for addressing the review!